### ISSUE TITLE
Marker getting placed slightly above the location

### ISSUE THREAD

**Issue by nitesh3691** [mapsplugin/cordova-plugin-googlemaps#1036]
Hi there,

I am developing a hybrid mobile app. I have plotted some markers on the map. The markers are getting displayed perfectly in ios but in android the marker placement is incorrect, its above the given location.

When we zoom in the map the difference is quite a big.

I have tried different version of map but so far no luck.

Any help would be appreciated.

Thanks,
Nitesh

**Comment by quiuquio** [comment]
Hi Nitesh, if you're using a custom icon for the marker, remember to set an anchor for it, or you'll get that funny behavior while zooming.

Something like this:

> var markerSizeScale = 1;
>       var markerSize = {
>         width: parseInt(40 * markerSizeScale),
>         height: parseInt(46 * markerSizeScale)
>       };
> 
>       map.addMarker({
>         position: position,
>         icon: {
>           'url': 'www/img/markers/tingparkmarker.png',
>           'draggable': true,
>           'size': markerSize,
>           'anchor': [markerSize.width / 2, markerSize.height] // Always add anchor to custom markers. Or the marker will move on zoom. [0,0] is top-left;
>         },
>         map: map.getMap(),
>         "infoWindowAnchor": [-110, 3] //[-go left, -go up]
>       }, .......

**Comment by nitesh3691** [comment]
Thanks quiuquio. It helped me to fix my marker on a location.

I have one more issue:

In my app I am continuously changing the location of a marker. On ios it is running perfectly on the track but on android sometimes it is moving away from the track.

Thanks once again

**Comment by quiuquio** [comment]
No problem! Could @wf9a5m75 comment if the need for the anchor on custom markers is expected or it is a bug?

For your other issue, how are you actually moving things? Some simple code sample would help.

**Comment by nitesh3691** [comment]
This is how I have created a marker

markerLocation = new plugin.google.maps.LatLng(marker[0].currentLat, marker[0].currentLong);

$rootScope.map.addMarker({
                    'position': markerLocation,
                    'title': markerName,
                    'icon': ENV.markerImage,
                    'size': markerSize,
                    'anchor': [markerSize.width / 2, markerSize.height]
                }, function(marker) {
                    $rootScope.myMarker = marker;
                });

Once my marker is created. I am calling a function with the below code to change its position

 $rootScope.myMarker.setPosition(markerLocation);

**Comment by quiuquio** [comment]
Your actual problem is still not very clear. Anyways it could be something related to $rootscope (avoid using it if you can, for example using services), or to the way you get the new markerLocations. Are the locations even correct?

I have a feeling that this might not be a problem related to the plugin itself.

**Comment by nitesh3691** [comment]
I am developing a tracking app ex: bus moving on a route. I have drawn a route on the map and trying to move my asset marker on it. I am getting the location from another app and here I am simply setting the position of the marker which I created previously and saved in rootscope.

$rootScope.myMarker.setPosition(markerLocation); // to set new location to the marker

I am getting new location from the service end.

On ios my asset is running perfectly on the track but in android build it is moving slightly away from the track.

**Comment by quiuquio** [comment]
A video would be nice at this point.
Is it erratically going out of track or there is a constant distance between track and marker? If it is the former, your coordinates are probably wrong, if it is the latter, there is probably an issue still related to marker anchor, which should probably be handled on another ticket. Also, try to make sure that the track is being drawn at the same place both on iOS and Android.

**Comment by nitesh3691** [comment]
Hi Quiuquio,

When I tried to run my app on different android devices I came across the issue that on some android devices the markers are not getting placed properly while on others they are.

I have attached screenshots for the same.

![android 5](https://cloud.githubusercontent.com/assets/10866403/17100143/515be56c-528a-11e6-88de-d4f400955f2b.png)

Here the markers are displayed incorrectly.

![android 5 5](https://cloud.githubusercontent.com/assets/10866403/17100159/6af28cce-528a-11e6-8133-c8fb16681b91.jpg)

Here the markers are up to the mark.

Both devices are currently have google map v9.32.1 and android 5.1.1.
